src/choices.c

```diff
@@ -83,7 +83,9 @@ merge(struct choice *front, struct choice *back)
 	c = &head;
 
 	while (front != NULL && back != NULL) {
-		if (front->score > back->score) {
+		if (front->score > back->score ||
+                (front->score == back->score &&
+                 strcmp(front->str, back->str) < 0)) {
```

Comment:
Yeah, I guess that's what strcmp(3) does. If `front->str` and `back->str` are definitely strings, I guess strcmp(3) is faster.